dasharo-security/secure-boot.robot

```diff
@@ -89,72 +92,67 @@ SBO002.002 UEFI Secure Boot (Windows 11)
-    [Documentation]    Keyword checks Secure Boot state in Linux.
-    ...    Returns True when Secure Boot is enabled
-    ...    and False when disabled.
-    ${out}=    Execute Linux Command    dmesg | grep secureboot
-    Should Contain Any    ${out}    disabled    enabled
-    ${sb_status}=    Run Keyword And Return Status
-    ...    Should Contain    ${out}    enabled
-    RETURN    ${sb_status}
-
-Check Secure Boot In Windows
-    [Documentation]    Keyword checks Secure Boot state in Windows.
-    ...    Returns True when Secure Boot is enabled
-    ...    and False when disabled.
-    ${out}=    Execute Command In Terminal    Confirm-SecureBootUEFI
-    Should Contain Any    ${out}    True    False
-    ${sb_status}=    Run Keyword And Return Status
-    ...    Should Contain    ${out}    True
-    RETURN    ${sb_status}
-
-Enable Secure Boot
-    Enter Setup Menu Tianocore
-    Enter Device Manager Submenu
-    Enter Secure Boot Configuration Submenu
-    Select Attempt Secure Boot Option
-    Save Changes And Reset    2
-
-Disable Secure Boot
-    Enter Setup Menu Tianocore
-    Enter Device Manager Submenu
-    Enter Secure Boot Configuration Submenu
-    Clear Attempt Secure Boot Option
-    Save Changes And Reset    2
+    Enable Secure Boot
+    Enable Custom Mode And Enroll Certificate    DB.cer
+    Enter UEFI Shell And Boot .EFI File    hello-valid-keys.efi    Hello, world!
+
+SBO004.001 Attempt to boot file without the key from Shell (firmware)
+    [Documentation]    This test verifies that Secure Boot blocks booting a file
+    ...    without a key.
+    Skip If    not ${SECURE_BOOT_SUPPORT}    SBO004.001 not supported
+    Skip If    not ${TESTS_IN_FIRMWARE_SUPPORT}    SBO004.001 not supported
+    Skip If    not ${TESTS_IN_UBUNTU_SUPPORT}    SBO004.001 not supported
+    Mount Image    ${PIKVM_IP}    not_signed.img
+    Power On
+    Enter UEFI Shell And Boot .EFI File    hello.efi    Access Denied
+
+SBO005.001 Attempt to boot file with the wrong-signed key from Shell (firmware)
+    [Documentation]    This test verifies that Secure Boot disallows booting
+    ...    a signed file with a wrong-signed key.
+    Skip If    not ${SECURE_BOOT_SUPPORT}    SBO005.001 not supported
+    Skip If    not ${TESTS_IN_FIRMWARE_SUPPORT}    SBO005.001 not supported
+    Skip If    not ${TESTS_IN_UBUNTU_SUPPORT}    SBO005.001 not supported
+    Mount Image    ${PIKVM_IP}    bad_keys.img
+    Power On
+    Enter UEFI Shell And Boot .EFI File    hello-bad-keys.efi    Access Denied
+
+SBO006.001 Reset Secure Boot Keys option availability (firmware)
+    [Documentation]    This test verifies that the Reset Secure Boot Keys
+    ...    option is available
+    Skip If    not ${SECURE_BOOT_SUPPORT}    SBO006.001 not supported
+    Skip If    not ${TESTS_IN_FIRMWARE_SUPPORT}    SBO006.001 not supported
+    Skip If    not ${TESTS_IN_UBUNTU_SUPPORT}    SBO006.001 not supported
+    Power On
+    Verify Reset Secure Boot Keys
```

Comment:
I would remove this KW and use directly as we do in other tests

```
    Enter Setup Menu Tianocore
    Enter Device Manager Submenu
    Enter Secure Boot Configuration Submenu
    Read From Terminal Until    Reset Secure Boot Keys
```

Reply:
Resolved in [21a0c1f](https://github.com/Dasharo/open-source-firmware-validation/pull/17/commits/21a0c1f4bc636ec5ee87d9cd923764a084c70656)